### DISCUSSION TITLE
Make osu!mania miss window constant across all ODs.

### DISCUSSION THREAD

**Discussion by Hydria** [ppy/osu#25913] [Ideas]
Right now in osu.Game.Rulesets.Scoring.Hitwindows.cs, we can see the HitResult.Miss is scaled from 188ms -> 158ms dependent on the OD. This doesn't make sense in terms of OD scaling, as it allows additional time for a player to not worry about the hitwindow, therefore negating the fact that the OD has _increased_. The biggest consequence of this is best shown in the image below:
![image 1](https://github.com/ppy/osu/assets/13346079/45b5b1b3-23a6-4d7f-aa19-11a5da204930)

If the HitResult.Miss is set as a constant, it doesn't allow hitwindow abuse with higher ODs, and should be scaled to _at least_ the OD0 hitwindow to keep consistency with how old hitwindows worked, which looks like so (for OD10):
![image 3](https://github.com/ppy/osu/assets/13346079/cc23e4ee-b0c5-4937-92ef-67e72c82c5e7)

I know this is possible to set as for osu!std, in osu.Game.Rulesets.Osu.Scoring.OsuHitWindows.cs, the miss window is forced at 400ms constant regardless of the OD.

Question for Dev Team: How feasible is this to implement, given that prior scores are set on a more forgiving hitwindow? Would you have to recalculate every score set?

Question for the community: Is 188ms fine for a miss hitwindow or should the hitwindow be higher, i.e. 200ms?

**Comment by bdach** [comment]
Just to clarify: is this a problem with lazer only, or is it a problem with stable too?

**Reply by Hydria** [reply]
It would affect both, yeah (in theory; stable doesn't actually display the miss hitwindow), which is why I understand that the feasibility of this is _fairly low_ going into it, but this is also the cornerstone towards better SR fixes down the line.

**Reply by bdach** [reply]
I don't know about long term but it's 99% not happening this year then.

**Reply by Hydria** [reply]
@bdach one followup question, do we actually have the replays stored to do score recalcs on _every single play_ for this case or would an alternate solution be in play for this?

**Reply by bdach** [reply]
no, recalculation of every single play is not possible, full stop.
